docs/secrets-and-passwords.md

```diff
@@ -4,72 +4,126 @@ _A documentation on secure management of secrets and passwords in projects_
-4. Best Practices for Security
-   - 4.1 Use of Strong Passwords
-5. Conclusion
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [1Password Vault for Secrets and Passwords](#2-1password-vault-for-secrets-and-passwords)
+   - [2.1 Vault Naming Conventions](#21-vault-naming-conventions)
+   - [2.2 Essential Vault Guidelines](#22-essential-vault-guidelines)
+3. [1Password Service Accounts](#3-1password-service-accounts)
+   - [3.1 Requesting Service Account Access](#31-requesting-service-account-access)
+   - [3.2 Storing Service Account Token](#32-storing-service-account-token)
+   - [3.3 Why Use Service Accounts?](#33-why-use-service-accounts)
+   - [3.4 Important Considerations](#34-important-considerations)
+4. [Using 1Passwords GitHub Actions](#4-using-1passwords-github-actions)
+   - [4.1 Integration](#41-integration)
+   - [4.2 Secrets Reference Syntax](#42-secrets-reference-syntax)
+5. [Best Practices](#5-best-practices)
+   - [5.1 Use of Strong Passwords](#51-use-of-strong-passwords)
+   - [5.2 Reusable Workflow for getting .env File from 1Password Vault](#52-reusable-workflow-for-getting-env-file-from-1password-vault)
+6. [Handling Leaked Secrets](#6-handling-leaked-secrets)
+7. [Organizational-Level Secrets Management](#7-organizational-level-secrets-management)
+8. [Conclusion](#8-conclusion)
 
 ---
 
 ## 1. Introduction
 
-Ensuring the security of secrets and passwords in a project is of paramount importance to safeguard sensitive information from unauthorized access. This documentation outlines best practices for managing secrets and passwords in a project environment using 1Password and GitHub.
+Ensuring the security of secrets and passwords in a project is of paramount importance to safeguard sensitive information from unauthorized access. Our goal is to ensure the integrity of our security-relevant data. At the same time, we want to offer a frustration-free way to use this data within our development processes. This documentation outlines best practices for managing secrets and passwords in a project environment using 1Password and GitHub.
+
+**[back to top](#table-of-contents)**
 
 ## 2. 1Password Vault for Secrets and Passwords
 
 All secrets and passwords of a project must be managed within 1Password. A dedicated vault for each project within 1Password facilitates organized and centralized storage of sensitive information. Access to this vault should be limited to a trusted few, adhering to the principle of "as many as necessary, as few as possible." The use of 1Password ensures that data is encrypted and protected.
 
-### 2.1 Naming Conventions:
+### 2.1 Vault Naming Conventions
 
-To maintain consistency and clarity, we suggest to use the following naming convention for our project vaults: `client-project`. Here's what it means:
+To maintain consistency and clarity, we suggest to use the following naming convention for our project vaults: `client_project`. Here's what it means:
```

Comment:
Why did you change from dash case to snake case? Is there a reason to prefer one over the other? I actually like dash case more but this is only a preference. If there is an actual reason why one is better, let's use the better one then.

Reply:
I like to use kebab-case for names and snake_case for hierarchies. In this case, "client" and "project" are each a level in the hierarchy with their own names. Usually names don't have underlines in them but dashes can occur and that would mess up the hierarchy.